extensions/sdk/src/Rally.ts

```diff
@@ -99,9 +99,49 @@ export class Rally {
       connectFirestoreEmulator(this._db, 'localhost', 8080);
     }
 
+    this.storeAttributionCodes()
+      .catch((err) => console.error("Could not store attribution codes:", err));
+
     onAuthStateChanged(this._auth, user => this.authStateChangedCallback(user));
   }
 
+  /**
+   * Attempt to fetch the attribution codes from the store page URL for this extension.
+   */
+  private async storeAttributionCodes() {
+    const attribution = await this.getAttributionCodes();
+    if (!(Object.keys(attribution).length === 0 && attribution.constructor === Object)) {
```

Comment:
I don't think this is quite the same... we want to know if this is an object with no keys (`{}`), since that's what `browser.storage.get()` will resolve with if there's no data.